c10/util/Exception.cpp

```diff
@@ -133,6 +133,10 @@ void warn(SourceLocation source_location, const std::string& msg, const bool ver
   ThreadWarningHandler::get_handler()->process(source_location, msg, verbatim);
 }
 
+void warn(SourceLocation source_location, const char* msg, const bool verbatim) {
+  ThreadWarningHandler::get_handler()->process(source_location, msg, verbatim);
```

Comment:
But doesn't this convert to a std::string, anyway?

Reply:
The point is that the std::string constructor only gets code generated once (in the Exception.cpp), rather than at every string literal call site.